artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+	ExcludedRepositories   *[]string `yaml:"excludedRepositories,omitempty" xml:"excludedRepositories>repositoryRef,omitempty"`
+	RetentionPeriodHours   *int      `yaml:"retentionPeriodHours,omitempty" xml:"retentionPeriodHours,omitempty"`
+	CreateArchive          *bool     `yaml:"createArchive,omitempty" xml:"createArchive,omitempty"`
+	Enabled                *bool     `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	ExcludeBuilds          *bool     `yaml:"excludeBuilds,omitempty" xml:"excludeBuilds,omitempty"`
+	ExcludeNewRepositories *bool     `yaml:"excludeNewRepositories,omitempty" xml:"excludeNewRepositories,omitempty"`
+	SendMailOnError        *bool     `yaml:"sendMailOnError,omitempty" xml:"sendMailOnError,omitempty"`
+	Precalculate           *bool     `yaml:"precalculate,omitempty" xml:"precalculate,omitempty"`
+}
+
+// Backups is an alias for a slice of Backup that can be
+// properly marshaled to/from YAML.
+type Backups []*Backup
+
+// MarshalYAML implements the yaml.Marshaller interface for Backups.
+func (b Backups) MarshalYAML() (interface{}, error) {
+	backupsMap := make(map[string]*Backup)
+	for _, backup := range b {
+		if *backup == (Backup{Key: backup.Key}) {
+			backupsMap[*backup.Key] = nil
+		} else {
+			backupsMap[*backup.Key] = backup
+		}
+	}
+
+	return backupsMap, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Backups.
+func (b *Backups) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	backupsMap := make(map[string]*Backup)
+	if err := unmarshal(backupsMap); err != nil {
+		return err
+	}
+
+	var backupsSlice Backups
+	for backupKey, backupConfig := range backupsMap {
+		if backupConfig == nil {
+			continue
+		}
+		backupConfig.Key = &backupKey
+		backupsSlice = append(backupsSlice, backupConfig)
+	}
+
+	*b = backupsSlice
+	return nil
+}
+
+// Indexer represents the Maven Indexer settings in Artifactory Services Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-Servicesconfiguration(Backups,MavenIndexer)
+type Indexer struct {
+	Enabled              *bool     `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	CronExp              *string   `yaml:"cronExp,omitempty" xml:"cronExp,omitempty"`
+	IncludedRepositories *[]string `yaml:"includedRepositories,omitempty" xml:"includedRepositories>repositoryRef,omitempty"`
+	Reset                *bool     `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (i Indexer) MarshalYAML() (interface{}, error) {
+	if i.Reset != nil && *i.Reset {
+		return nil, nil
+	}
+	return i, nil
+}
+
+// RepoLayout represents a Repository Layout setting in Artifactory.
+// This is undocumented in YAML Configuration File.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/Repository+Layouts
+type RepoLayout struct {
+	Name                             *string `yaml:"-" xml:"name,omitempty"`
```

Comment:
I noticed we're not allowing this field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
Similar to [comment](https://github.com/target/go-arty/pull/57#discussion_r436119180).